src/overview/actions.js

```diff
@@ -257,12 +263,61 @@ export const showTags = index => (dispatch, getState) => {
 }
 
 export const filterTag = tag => (dispatch, getState) => {
-    const query = selectors.query(getState())
+    const state = getState()
+    const query = selectors.query(state)
+    const filterTags = selectors.filterTags(state)
+    const filterStatus = selectors.showFilter(state)
+
     const transformedTag = `#${tag.split(' ').join('+')} `
 
-    const newQuery = query.includes(transformedTag)
+    const newQuery = query.includes(transformedTag.slice(0, -1))
         ? query.replace(transformedTag, '') // Either remove it, if already there
         : transformedTag + query // or prepend it, if not there
 
     dispatch(setQuery(newQuery))
+
+    if (
```

Comment:
As it's a bit complicated because the states have to be synced, it would be good to put some comments in here explaining what's going on and why.